drivers/mcp23x17/mcp23x17.c

```diff
@@ -0,0 +1,622 @@
+/*
+ * Copyright (C) 2021 Gunar Schorcht
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     drivers_mcp23x17
+ * @brief       Device driver implementation for Microchip MCP23x17 I/O expanders
+ * @author      Gunar Schorcht <gunar@schorcht.net>
+ * @file
+ */
+
+#include <stdint.h>
+
+#include "assert.h"
+#include "periph/i2c.h"
+#include "ztimer.h"
+
+#include "mcp23x17.h"
+#include "mcp23x17_regs.h"
+
+#if IS_USED(MODULE_MCP23X17_IRQ)
+#include "event/thread.h"
+#endif
+
+#define ENABLE_DEBUG 0
+#include "debug.h"
+
+#if ENABLE_DEBUG
+
+#include <string.h>
+
+#define DEBUG_DEV(m, d, ...) \
+        DEBUG("[mcp23x17] %s dev=%" PRIxPTR " addr=%02x: " m "\n", \
+              __func__, (unsigned int)d, d->params.addr, ## __VA_ARGS__)
+
+#else /* ENABLE_DEBUG */
+
+#define DEBUG_DEV(f, d, ...)
+
+#endif /* ENABLE_DEBUG */
+
+#define _ADDR       (MCP23X17_BASE_ADDR + dev->params.addr)
+
+#if IS_USED(MODULE_MCP23X17_SPI)
+#define _SPI_DEV    (dev->params.if_params.spi.dev)
+#define _SPI_CS     (dev->params.if_params.spi.cs)
+#define _SPI_CLK    (dev->params.if_params.spi.clk)
+#endif
+
+#if IS_USED(MODULE_MCP23X17_I2C)
+#define _I2C_DEV    (dev->params.if_params.i2c.dev)
+#endif
+
+#if IS_USED(MODULE_MCP23X17_IRQ)
+
+#if IS_USED(MODULE_MCP23X17_IRQ_MEDIUM)
+#define  MCP23X17_EVENT_PRIO    EVENT_PRIO_MEDIUM
+#elif IS_USED(MODULE_MCP23X17_IRQ_HIGHEST)
+#define  MCP23X17_EVENT_PRIO    EVENT_PRIO_HIGHEST
+#endif
+
+/* interrutp service routine for IRQs */
```

Comment:
```suggestion
/* interrupt service routine for IRQs */
```

This is the typo the static tests were complaining about. Odd that the workflow didn't catch it before?

Reply:
This typo is still here 👀

Reply:
:sunglasses: